### PR TITLE
Fix JSON property deserialization

### PR DESCRIPTION
**Pull request by hepsimo**
Resolves:
DACT-134

- incorrect naming preventing correct deserialization
  - snake/camel case conversion not configured properly
  - non-standard extra underscores before numeral in address_line_1/2
- add appropriate Jackson JSON annotations to make use of builders

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/officerfiling/api/controller/OfficerFilingControllerImplIT.java [hunk 1]
```diff
@@ -1,21 +1,29 @@
 package uk.gov.companieshouse.officerfiling.api.controller;
```
Comment (PhilipGlover) [inline src/test/java/uk/gov/companieshouse/officerfiling/api/controller/OfficerFilingControllerImplIT.java:1]: this test doesn't compile

**Comment by PhilipGlover** [comment]
have submitted:
```    "reference_officer_id" :"off id 789",
        "residential_address": {
            "address_line_1": "string",
            "address_line_2": "string",
            "care_of": "string"
        }
```
and correctly stored in db:
```{
    "_id" : ObjectId("63329fda0699894a4820301e"),
    "name" : "bob1",
    ...
    "resigned_on" : ISODate("2022-09-20T00:00:00.000Z"),
    "residential_address" : {
        "address_line_1" : "string",
        "address_line_2" : "string",
        "care_of" : "string"
    },
    "_class" : "uk.gov.companieshouse.officerfiling.api.model.entity.OfficerFiling"
}
```

- this PR resolves defect DACT-134 - to be added to descriptions

File: src/test/java/uk/gov/companieshouse/officerfiling/api/controller/OfficerFilingControllerImplIT.java (hunk 1 above)
Comment (hepsimo) [reply src/test/java/uk/gov/companieshouse/officerfiling/api/controller/OfficerFilingControllerImplIT.java:1]: Oops, i'll fix these

**Review by PhilipGlover — APPROVED** [review]
- tests all green now
- re-submitted my json and stored in db as expected

**Pull request merged**